language/src/regent/parser.t

```diff
@@ -1375,6 +1375,21 @@ function parser.expr_prefix(p)
       has_parens = false,
     }
 
+  elseif p:nextif("__import_cross_product") then
+    local start = ast.save(p)
+    local partitions = terralib.newlist()
+    p:expect("(")
+    local list = parser.expr_list(p)
+    p:expect(")")
+    return ast.unspecialized.expr.ImportCrossProduct {
+      partitions = list:sub(1, #list - 2),
+      colors = list[#list - 1],
+      value = list[#list],
```

Comment:
We need a check on `#list` prior to doing this, since we can't assume the user has provided a valid number of arguments.